### PR TITLE
fix transpose bug

### PR DESCRIPTION
**Pull request by lucienwang1009**
when transpose's output is graph output, skip it.
@zhijxu-MS Please have a look after your PR about applying optimization in unittests check in.

### CODE REVIEW

File: tf2onnx/optimizer/transpose_optimizer.py [hunk 1]
```diff
@@ -252,6 +252,9 @@ def _switch_transpose_and_node(self, node, trans):
     # if return value is True, then it means Transpose is handled as designed
     # otherwise, it means that we skip handling since it is not in our support set
     def _handle_nhwc_tranpose(self, trans):
+        if trans.output[0] in self._g.outputs:
+            log.debug("%s connects to graph outputs", trans.output[0])
+            return False
```
Comment (nbcsm) [inline tf2onnx/optimizer/transpose_optimizer.py]: add a test to cover this case?

Comment (lucienwang1009) [reply tf2onnx/optimizer/transpose_optimizer.py]: need f850cbe15a3a3e928274e8d94400d756846baf7f

Comment (nbcsm) [reply tf2onnx/optimizer/transpose_optimizer.py]: yep

Comment (nbcsm) [reply tf2onnx/optimizer/transpose_optimizer.py]: @lucienwang1009, a simple case will be enough, tks.

**Pull request merged**